server/src/controllers/surveyController.test.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import CosmosClient from '../databases/cosmosClient';
+import SurveyDBHandler from '../databases/handlers/surveyDBHandler';
+import { storeSurveyResult } from '../controllers/surveyController';
+import { getServerConfig } from '../utils/getConfig';
+
+jest.mock('../databases/cosmosClient');
+
+describe('surveyResultController', () => {
+  let response;
+  let next;
+
+  beforeEach(() => {
+    process.env.VV_COSMOS_DB_ENDPOINT = 'https://testinghost/';
+    response = {
+      send: jest.fn().mockReturnThis(),
+      status: jest.fn().mockReturnThis()
+    } as any;
+    next = jest.fn();
+  });
+
+  afterEach(() => {
+    delete process.env.VV_COSMOS_DB_ENDPOINT;
+    jest.resetAllMocks();
+  });
+
+  test('Should success on upsert', async () => {
+    const inputData: any = {
+      sessionId: 'test_session_id',
+      callId: 'test_call_id',
+      acsUserId: 'test_acs_user_id',
+      response: true
+    };
+    const request: any = { body: inputData };
+
+    const config = getServerConfig();
+    let cosmosClient;
+
+    if (config.cosmosDb) {
```

Comment:
To keep separation of concerns and to keep the spirit of unit testing, I think we can assume here that cosmosDb is defined, and just create a mock config (no need to call `getServerConfig()` because then these become integration tests.).

IMO testing whether cosmosDb is defined or not is more suited for app.test.ts, not here. In fact you already are covering that case with this test case `check if /api/surveyResults route is open without cosmosDb configs`.